packages/scene-composer/src/components/Tree/TreeItem.tsx

```diff
@@ -23,30 +25,34 @@ export interface TreeItemProps extends TreeItemInnerProps, ComponentPropsWithRef
 
 const TreeItemInner: FC<TreeItemInnerProps> = ({
   children,
-  onSelected = /* istanbul ignore next */ () => {},
   selectable = true,
   selected = false,
   className = '',
   onActivated = () => {},
 }) => {
+  const sceneComposerId = useSceneComposerId();
+
   const toggle = useCallback(
     (e) => {
-      onSelected(!selected, e);
+      (children as any)?.filter((child) => {
+        return child ? useStore(sceneComposerId).getState().setSelectedSceneNodeRef(child.props?.objectRef) : null;
```

Comment:
use hook inside hook is anti-pattern

Reply:
nvm, this pattern is introduced recently to fix performance issue. I will look into if we can fix the performance issue from the useStore hook side instead of calling hook inside hook